### PR TITLE
Integrated Wolfram Alpha API to enable mathematical expression evaluation 

### PR DESCRIPTION
**Pull request by Moses-Mk**
Integrated Wolfram Alpha API to enable mathematical expression evaluation and plot display in the calculator. Refined plot handling to ensure only valid images are shown, and added error handling for cases with no plots or results.

Closes: 

- **Title**:  Calculator Under API
- **Your Name**: Moses Makuei Jiet
- **Idenitfy yourself**: Contributor

### Describe the add-ons or changes you've made 📃

- Integrated Wolfram Alpha API to perform mathematical calculations and generate plots.
- Improved error handling for queries with no results or plots.
- Streamlined result display by preventing redundancy in text and image outputs.

### Checklist: ☑️
- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/Contributing.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/Jarvis/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

Screenshot 2024-10-06 135644

### CODE REVIEW

File: src/apps/pages/programs/ApiPrograms/calculator.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            # Perform the Wolfram Alpha query
+            result_pods = calculate_expression(query, WOLFRAM_API_KEY)
+            
+            # Display the results
+            display_results(result_pods)
+        else:
+            st.error("Please enter a valid query!", icon="🚨")
+
+# Run the app
+if __name__ == "__main__":
```
Comment (Avdhesh-Varshney) [inline src/apps/pages/programs/ApiPrograms/calculator.py:80]: Remove it and don't call the function. It is automatically called.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
After removing it, attach the working video here also.

File: src/apps/pages/programs/ApiPrograms/calculator.py (hunk 1 above)
Comment (Moses-Mk) [reply src/apps/pages/programs/ApiPrograms/calculator.py]: I have addressed the changes sir.. You can now merge my request or update me if there is anything more i can do. Thanks

Here below is the working video:
 [Watch the video on Google Drive](https://drive.google.com/file/d/1LN5O8gFw5bRETqi4YRVaI4Mn3c8ldAsL/view?usp=sharing)

**Review by Avdhesh-Varshney — APPROVED** [review]

**Review by Zedoman — APPROVED** [review]

**Comment by Ganesh2409** [comment]
hii @Moses-Mk can u please help me with the project set up ?

**Pull request merged**